### PR TITLE
option to create an object only once in multi-iteration jobs

### PR DESCRIPTION
**Pull request by mukrishn**
## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [x] Optimization
- [ ] Documentation Update

## Description

In certain cases on creating cluster wide resources like clusterroles, operator spec CRDs, network spec, they would not have to run for every iteration although it is idempotent, this additional option let us attempt just once during the  iteration within a job. 
We have many of such scenarios with ICNI implementaion like serviceaccount, SRIOV specs, ICNI CRDs are required to run only once within a job.

example object, when jobIteration is > 1
```
      - objectTemplate: serviceaccount.yml
        replicas: 1
        runOnce: true

      - objectTemplate: pod.yml
        replicas: 1
```
## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Review by venkataanil — APPROVED** [review]
Agree, needed feature.

File: pkg/burner/create.go [hunk 1]
```diff
@@ -149,7 +149,15 @@ func (ex *Executor) RunCreateJob(iterationStart, iterationEnd int, waitListNames
 				"kube-burner-runid": ex.runid,
 			}
 			ex.objects[objectIndex].labelSelector = labels
-			ex.replicaHandler(labels, obj, ns, i, &wg)
+			if obj.RunOnce {
+				if i == 0 {
```
Comment (rsevilla87) [inline pkg/burner/create.go:153]:
```suggestion
			if obj.RunOnce && i == 0 {
				if i == 0 {
```

File: pkg/config/types.go [hunk 2]
```diff
@@ -92,6 +92,8 @@ type Object struct {
 	Wait bool `yaml:"wait" json:"wait"`
 	// WaitOptions define custom behaviors when waiting for objects creation
 	WaitOptions WaitOptions `yaml:"waitOptions" json:"waitOptions,omitempty"`
+	// Run Once to create the object only once incase of multiple iterative jobs
+	RunOnce bool `yaml:"runonce" json:"runonce,omitempty"`
```
Comment (rsevilla87) [inline pkg/config/types.go:96]: Can you add this param to the docs?

File: pkg/burner/create.go [hunk 3]
```diff
@@ -149,7 +149,15 @@ func (ex *Executor) RunCreateJob(iterationStart, iterationEnd int, waitListNames
...
+					// this executes only once during the first iteration of an object
+					log.Infof("RunOnce set to %s, so creating object once", obj.ObjectTemplate)
```
Comment (rsevilla87) [inline pkg/burner/create.go:155]: I'd remove this log message, is very verbose, it prints the entire template

**Comment by rsevilla87** [comment]
I normally recommend to use a different job with just one iteration to create this kind of objects, but this feature can be useful for other scenarios as well

File: pkg/burner/create.go (hunk 1 above)
Comment (mukrishn) [reply pkg/burner/create.go:153]: I had this first but it didn't work as expected, it goes in to `if` condition during first iteration and goes to `else` for all other iteration irrespective of `RunOnce`

File: pkg/burner/create.go (hunk 3 above)
Comment (mukrishn) [reply pkg/burner/create.go:155]: @rsevilla87 I can make it to debug, but this prints just this only once per object, 
```
time="2024-01-03 17:53:46" level=info msg="Creating pods.yml object only once" file="create.go:155"
```
I changed the message later but was using the same variable `obj.ObjectTemplate`

File: pkg/config/types.go (hunk 2 above)
Comment (rsevilla87) [inline pkg/config/types.go:96]: Just one last nit, can you rename this new param to `runOnce` in order to follow camelcase semanthics as the other params?, thanks!

**Review by rsevilla87 — APPROVED** [review]
lgtm

**Pull request merged**